pkg/cloud_provider/metadata/metadata.go

```diff
@@ -45,9 +53,15 @@ func NewMetadataService() (Service, error) {
 		return nil, fmt.Errorf("failed to get project: %v", err)
 	}
 
+	network, err := getNetwork()
```

Comment:
If the controller runs outside of the user's project, is this going to work?

Reply:
@saikat-royc I can create several projects in the mattcary.info org we can use to test out various network configurations. Let's sync offline on that.